src/awx_plugins/credentials/aws_assumerole.py

```diff
@@ -0,0 +1,165 @@
+"""This module provides integration with AWS AssumeRole functionality."""
+
+import datetime
+import hashlib
+
+import boto3
+from botocore.exceptions import ClientError
+
+from .plugin import CredentialPlugin, translate_function as _
+
+
+_aws_cred_cache = {}
+
+
+assume_role_inputs = {
+    'fields': [
+        {
+            'id': 'access_key',
+            'label': _('AWS Access Key'),
+            'type': 'string',
+            'secret': True,
+            'help_text': _(
+                'The optional AWS access key for'
+                'the user who will assume the role.',
+            ),
+        },
+        {
+            'id': 'secret_key',
+            'label': 'AWS Secret Key',
+            'type': 'string',
+            'secret': True,
+            'help_text': _(
+                'The optional AWS secret key for the'
+                'user who will assume the role.',
+            ),
+        },
+        {
+            'id': 'external_id',
+            'label': 'External ID',
+            'type': 'string',
+            'help_text': _(
+                'The optional External ID which will'
+                'be provided to the assume role API.',
+            ),
+        },
+    ],
+    'metadata': [
+        {
+            'id': 'identifier',
+            'label': 'Identifier',
+            'type': 'string',
+            'help_text': _(
+                'The name of the key in the assumed AWS role'
+                'to fetch [AccessKeyId | SecretAccessKey | SessionToken].',
+            ),
+        },
+    ],
+    'required': [
+        'role_arn',
+    ],
+}
+
+
+def aws_assumerole_getcreds(
+        access_key: str | None,
+        secret_key: str | None,
+        role_arn: str | None,
+        external_id: int,
+) -> dict:
+    """Return the credentials for use.
+
+    :param access_key: The AWS access key ID.
+    :type access_key: str
+    :param secret_key: The AWS secret access key.
+    :type secret_key: str
+    :param role_arn: The ARN received from AWS.
+    :type role_arn: str
+    :param external_id: The external ID received from AWS.
+    :type external_id: int
+    :returns: The credentials received from AWS.
+    :rtype: dict
+    :raises ValueError: If the client response is bad.
+    """
+    explicit_credentials_empty = not access_key and not secret_key
+    credential_kwargs = {} if explicit_credentials_empty else {
+        # EE creds are read from the env
+        'aws_access_key_id': access_key,
+        'aws_secret_access_key': secret_key,
+    }
+    connection = boto3.client(service_name='sts', **credential_kwargs)
+    try:
+        response = connection.assume_role(
+            RoleArn=role_arn,
+            RoleSessionName='AAP_AWS_Role_Session1',
+            ExternalId=external_id,
+        )
+    except ClientError as ce:
+        raise ValueError(f'Got a bad client response from AWS: {ce.message}.')
```

Comment:
It's usually better to chain the exceptions explicitly and chose more verbose var names:
```suggestion
    except ClientError as client_err:
        raise ValueError(
            f'Got a bad client response from AWS: {ce.message}.',
        ) from client_err
```